license_manager/apps/subscriptions/models.py

```diff
@@ -1512,10 +1512,23 @@ def get_licenses_exceeding_purge_duration(cls, date_field_to_compare, **kwargs):
             date_field: localized_utcnow() - models.F(duration_before_purge_field),
         })
 
-        return License.objects.filter(**kwargs).select_related(
+        # ordered by primary key for stable pagination, which we'll rely on
+        # below to generate batches of result sets.
+        base_queryset = License.objects.filter(**kwargs).select_related(
             'subscription_plan',
             'subscription_plan__customer_agreement',
-        )
+        ).order_by('pk')
+
+        queryset = base_queryset[:batch_size]
+        offset = 0
+        while queryset.exists():
+            yield queryset
+            # queryset.last() doesn't work here because we've already sliced
+            # loop through so we eventually grab the last one
+            for item in queryset:
+                offset = item.pk
```

Comment:
Can't you index a sliced queryset as long as it's already evaluated? Not tested:
```suggestion
            offset = queryset[-1]
```